compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/PointerPositionUpdater.skiko.kt

```diff
@@ -16,92 +16,39 @@
 
 package androidx.compose.ui
 
-import androidx.compose.ui.input.pointer.HitPathTracker
-import androidx.compose.ui.input.pointer.PointerEventType
-import androidx.compose.ui.input.pointer.PointerInputEvent
 import kotlin.js.JsName
 
+/**
+ * Updater of the current position of pointer.
+ *
+ * If something happened with Compose content (it relayouted), we need to send an
+ * event to it with the latest pointer position. Otherwise, the content won't be updated
+ * by the actual relative position of the pointer.
+ *
+ * For example, it can be needed when we scroll content without moving the pointer, and we need
+ * to highlight the items under the pointer.
+ */
 internal class PointerPositionUpdater(
```

Comment:
It was extracted because in reality they dealing with different problems. 

PointerPositionUpdater informs Compose content about the updated position after relayout. Now it informs via sending synthetic events, but that just one of the options.

SyntheticEventSender encapsulates generation of synthetic events based on the previous events. Besides, in the next PR I add generation of Press/Release, which doesn't fit PointerPositionUpdater